source/server/config/network/http_connection_manager.cc

```diff
@@ -43,9 +50,18 @@ HttpConnectionManagerFilterConfigFactory::createFilterFactory(const Json::Object
                                                                       context.threadLocal());
           });
 
-  std::shared_ptr<HttpConnectionManagerConfig> http_config(
-      new HttpConnectionManagerConfig(config, context, *date_provider));
-  return [http_config, &context, date_provider](Network::FilterManager& filter_manager) -> void {
+  std::shared_ptr<Router::RouteConfigProviderManager> route_config_provider_manager =
+      context.singletonManager().getTyped<Router::RouteConfigProviderManager>(
+          route_config_provider_manager_singleton_name, [&context] {
+            return std::make_shared<Router::RouteConfigProviderManagerImpl>(
+                context.runtime(), context.dispatcher(), context.random(), context.localInfo(),
+                context.threadLocal());
+          });
+
+  std::shared_ptr<HttpConnectionManagerConfig> http_config(new HttpConnectionManagerConfig(
+      config, context, *date_provider, *route_config_provider_manager));
```

Comment:
Are we losing the reference count here? The singleton manager only has a weak pointer, when `route_config_provider_manager` goes out of scope above, it will delete the `RouteConfigProviderManager`.